packages/edit-navigation/src/components/layout/use-navigation-block-with-name.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * WordPress dependencies
+ */
+import { useEffect } from '@wordpress/element';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { BlockControls } from '@wordpress/block-editor';
+import { addFilter, removeFilter } from '@wordpress/hooks';
+import {
+	__experimentalEditInPlaceControl as EditInPlaceControl,
+	ToolbarGroup,
+} from '@wordpress/components';
+
+export default function useNavigationBlockWithName( menuId ) {
+	const menu = useSelect( ( select ) => select( 'core' ).getMenu( menuId ), [
+		menuId,
+	] );
+
+	const { saveMenu } = useDispatch( 'core' );
+
+	useEffect( () => {
+		if ( menu ) {
+			const withMenuName = createHigherOrderComponent(
+				( BlockEdit ) => ( props ) => {
+					if ( props.name !== 'core/navigation' ) {
+						return <BlockEdit { ...props } />;
+					}
+					return (
+						<>
+							<BlockEdit { ...props } />
+							<BlockControls>
+								<ToolbarGroup>
+									<EditInPlaceControl
+										label={ menu?.name ?? '(untitled menu)' }
```

Comment:
what about internalization here?

Reply:
Good catch we should internationalize this label!